common/remote/hooks/azure/azurefile.cpp

```diff
@@ -0,0 +1,827 @@
+{
+    UNIMPLEMENTED_X("AzureFileAppendBlobWriteIO::appendFile");
+    return 0;
+}
+
+offset_t AzureFileAppendBlobWriteIO::size()
+{
+    DBGLOG("Warning: Size (%" I64F "u) requested on output IO", offset);
+    return offset;
+}
+
+size32_t AzureFileAppendBlobWriteIO::write(offset_t pos, size32_t len, const void * data)
+{
+    if (len)
+    {
+        if (offset != pos)
+            throw makeStringExceptionV(100, "Azure file output only supports append.  File %s %" I64F "u v %" I64F "u", file->queryFilename(), pos, offset);
+
+        stats.ioWrites++;
+        stats.ioWriteBytes += len;
+        CCycleTimer timer;
+        {
+            file->appendToAppendBlob(len, data);
+            offset += len;
+        }
+        stats.ioWriteCycles += timer.elapsedCycles();
+    }
+    return len;
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+
+AzureFileBlockBlobWriteIO::AzureFileBlockBlobWriteIO(AzureFile * _file) : AzureFileWriteIO(_file)
+{
+    file->createAppendBlob();
+}
+
+void AzureFileBlockBlobWriteIO::close()
+{
+    file->commitBlockBlob();
+}
+
+offset_t AzureFileBlockBlobWriteIO::appendFile(IFile *file, offset_t pos, offset_t len)
+{
+    UNIMPLEMENTED_X("AzureFileBlockBlobWriteIO::appendFile");
+    return 0;
+}
+
+size32_t AzureFileBlockBlobWriteIO::write(offset_t pos, size32_t len, const void * data)
+{
+    if (len)
+    {
+        assertex(offset == pos);
+        file->appendToBlockBlob(len, data);
+        offset += len;
+    }
+    return len;
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+
+void throwStorageError(const azure::storage_lite::storage_error & error)
+{
+    unsigned errCode = atoi(error.code.c_str());
+    throw makeStringExceptionV(errCode, "Azure Error: %s, %s", error.code.c_str(), error.code_name.c_str());
+}
+
+template <typename RESULT>
+void checkError(const RESULT & result)
+{
+    if (!result.success())
+        throwStorageError(result.error());
+}
+
+static std::shared_ptr<azure::storage_lite::blob_client> getClient(const char * accountName, const char * key)
+{
+    StringBuffer accountTemp;
+    if (!accountName)
+        accountName = getenv("AZURE_ACCOUNT_NAME");
+    if (!key)
+        key = getenv("AZURE_ACCOUNT_KEY");
+
+    std::shared_ptr<azure::storage_lite::storage_credential> cred = nullptr;
+    try
+    {
+        cred = std::make_shared<azure::storage_lite::shared_key_credential>(accountName, key);
+    }
+    catch (const std::exception & e)
+    {
+        IException * error = makeStringExceptionV(1234, "Azure access: %s", e.what());
+        throw error;
+    }
+
+    std::shared_ptr<azure::storage_lite::storage_account> account = std::make_shared<azure::storage_lite::storage_account>(accountName, cred, /* use_https */ true);
+    return std::make_shared<azure::storage_lite::blob_client>(account, 10);
```

Comment:
what it the '10' constant?

Reply:
Very good question - I can't remember.  I'll document it.